File: sources/dart/09_fib_recursive/main.dart
// task 09 fib_recursive — expected output: 102334155
// build: dart compile exe -o prog main.dart (aot; jit has no build step)    run: ./prog (aot) | dart main.dart (jit)

int fib(int n) {
  if (n < 2) {
    return n;
  }
  return fib(n - 1) + fib(n - 2);
}

void main() {
  print(fib(40));
}

File: sources/scala/09_fib_recursive/main.scala
// task 09 fib_recursive — expected output: 102334155
// build: scalac -release 17 main.scala    run: scala Main

object Main {
  private def fib(n: Long): Long =
    if (n < 2L) n else fib(n - 1L) + fib(n - 2L)

  def main(args: Array[String]): Unit = {
    println(fib(40L))
  }
}
